proposals/0000-bytearray-literals.rst

```diff
@@ -0,0 +1,189 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+
+.. highlight:: haskell
+
+This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/N>`_.
+
+.. contents::
+
+ByteArray# literals
+===================
+
+This is a proposal to introduce ``ByteArray#`` literals. The user
+would be able to write:
+
+.. code-block:: haskell
+
+  "Literals"#         -- Addr# (Modified UTF-8)
+  "\xef\xbb\xbf"utf8# -- Addr# (UTF-8)
```

Comment:
How to write `Addr#` of binary data, which isn't valid UTF-8 stream?

Without a way, this does break existing programs.

Reply:
Oh, I had forgotten that `Addr#` literals offer a combination of raw bytes and modified UTF-8. I'll update the proposal to mention that this will stay the same. And I'll add `bin` as a blessed encoding so that `"\xef\xbb\xbf"bin#` and ``"\xef\xbb\xbf"bin##` are allowed. This encoding will only allow hexadecimal-encoded bytes like the ones in the example.